library/init/data/nat/gcd.lean

```diff
@@ -9,7 +9,7 @@ prelude
 import init.data.nat.lemmas init.meta.well_founded_tactics
 
```

Comment:
I'm not sure if you should remove the gcd.  `nat.gcd` and `int.gcd` have an efficient VM implementations using GMP.

One option is to keep the gcd implementation as meta constants, and then use `vm_override` in mathlib to hook up to fast implementation:
```lean
meta constant nat.gcd_gmp : ℕ → ℕ → ℕ

-- in mathlib:
@[vm_override nat.gcd_gmp]
def nat.gcd : ℕ → ℕ → ℕ := ...
```

Reply:
@digama0 I thought you said it could go. I'll let you two figure this out.

Reply:
Ok, I'm putting the defs back

Reply:
Oh, I didn't know that these were vm overridden. If so, then they should stay. But AFAIR they don't require much automation, although the def has to be stated a bit differently for the decreasing proof to work.

Reply:
@digama0 I don't follow. Is there something that should be changed? The CI seems happy.

Reply:
I'm not sure what the latest version of this is, but you've probably fixed it. You asked about it earlier and I said to shift the `| (succ x) y := ...` cases to `| (succ x) := \lam y, ...` so that lean gets the recursion domain right.

Reply:
@digama0 Aah, that was on a different branch. I decided not to pursue that here.
Anyways, that settles this discussion, I think.

Reply:
What happened to removing `default_dec_tac`?

Reply:
No, it turns out that it isn't needed.

Reply:
It might still be a good idea. But it can be a different PR.